providers/os/resources/usb/parser_plist.go

```diff
@@ -32,6 +34,11 @@ type USBDevice struct {
 func ParseMacosIORegData(data any, devices *[]USBDevice) {
 	// Process the data based on its type
 	switch v := data.(type) {
+	case plist.Data:
+		// root object, we need to convert it to map[string]any
+		obj := data.(plist.Data)
+		// typecase so that we reach case map[string]any
+		ParseMacosIORegData(map[string]any(obj), devices)
```

Comment:
the name here mentions macos but is used for usb, maybe we can rename it to be a bit more generic?